File: DSA/Pragya_Agarwal/Day2_Question2.py
leaderboard = [100, 90, 90, 80]
player_scores = [70, 80, 105]

unique_leaderboard = []
for score in leaderboard:
    if score not in unique_leaderboard:
        unique_leaderboard.append(score)

ranks = []

for score in player_scores:
    rank = 1
    previous_score = None
    for lb_score in unique_leaderboard:
        if score < lb_score and lb_score != previous_score:
            rank += 1
            previous_score = lb_score
        else:
            break
    ranks.append(rank)

print(ranks) 

File: DSA/Rakshit-Singh/Day 2/Day_2_Question_2.java
class Day_2_Question_2
{
    public int[] leaderboard_rank(int[] leaderboard,int player_scores[]){
        int p_len=player_scores.length,l_len=leaderboard.length;
        int ranks[]=new int[p_len];
        
        for(int x=0;x<p_len;x++){
            int rank=1;
            for(int y=0;y<l_len;y++){
                if(player_scores[x]>=leaderboard[y]){
                    break;
                }
                else if(y+1!=l_len && leaderboard[y]==leaderboard[y+1]){
                    continue;
                }
                rank++;
            }
            ranks[x]=rank;
        }
        return ranks;
    }
}